src/server/main_service.cc

```diff
@@ -678,6 +678,12 @@ optional<Transaction::MultiMode> DeduceExecMode(ExecEvalState state,
     for (const auto& scmd : exec_info.body) {
       transactional |= scmd.Cid()->IsTransactional();
       contains_global |= scmd.Cid()->opt_mask() & CO::GLOBAL_TRANS;
+
+      // We can't run no-key-transactional commands in lock-ahead mode currently,
+      // because it means we have to schedule on all shards
+      if (scmd.Cid()->opt_mask() & CO::NO_KEY_TRANSACTIONAL)
```

Comment:
I think the flag you check should be NO_KEY_TX_SPAN_ALL

Reply:
No, because currently transactions that don't "span" all shards, choose the first shard by default (which can be inactive), so they will fail as well if the tx was not scheduled on shard 0